packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -601,6 +601,27 @@ export abstract class BaseService extends Resource
     return cloudmapService;
   }
 
+  /**
+   * Associates this service with a CloudMap service
+   */
+  public associateCloudMapService(options: AssociateCloudMapServiceOptions): void {
+    const service = options.service;
+
+    const { containerName, containerPort } = determineContainerNameAndPort({
+      taskDefinition: this.taskDefinition,
+      dnsRecordType: service.dnsRecordType,
+      container: options.container,
+      containerPort: options.containerPort,
+    });
+
+    // add Cloudmap service to the ECS Service's serviceRegistry
+    this.addServiceRegistry({
+      arn: service.serviceArn,
+      containerName,
+      containerPort,
```

Comment:
I think it'd be helpful to see if `this.serviceRegistries` is empty, or add a validation within `addServiceRegistry` -- otherwise the ECS service will fail its validations, since it can have at most 1 service registry.

Reply:
@SoManyHs Sounds good to me. This got me a while back and I had already forgotten. Haha.